File: app/src/main/java/com/example/myapplication2/myAdapter.java
package com.example.myapplication2;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import de.hdodenhof.circleimageview.CircleImageView;

public class myAdapter extends RecyclerView.Adapter<myAdapter.Myviewholder> {

    private List<Model> playerList;
    private clickInterface clickInterface;

    public myAdapter(List<Model> playerList,clickInterface clickInterface) {
        this.playerList = playerList;
        this.clickInterface = clickInterface;
    }

    @NonNull
    @Override
    public Myviewholder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        View view= LayoutInflater.from(parent.getContext()).inflate(R.layout.single_itemlayout,parent,false);

        return  new Myviewholder(view);

    }

    @Override
    public void onBindViewHolder(@NonNull Myviewholder holder, int position) {

        holder.imageView.setImageResource(playerList.get(position).getPosition());
        holder.textView.setText(playerList.get(position).getName());


    }

    @Override
    public int getItemCount() {
        return playerList.size();
    }

    public class Myviewholder extends RecyclerView.ViewHolder {

        private CircleImageView imageView;
        private TextView textView;
        private TextView type;


        public Myviewholder(@NonNull View itemView) {
            super(itemView);
            imageView=itemView.findViewById(R.id.singleImageId);
            textView=itemView.findViewById(R.id.singleNameId);

            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    clickInterface.onItemClick(getAdapterPosition());
                }
            });

            itemView.setOnLongClickListener(new View.OnLongClickListener() {
                @Override
                public boolean onLongClick(View view) {
                    clickInterface.onLongClick(getAdapterPosition());
                    return true;
                }
            });

        }
    }
}

File: Kotlin2/app/src/main/java/com/example/kotlin2/MyAdapter.kt
package com.example.kotlin2


import android.os.Parcel
import android.os.Parcelable
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.AdapterView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import de.hdodenhof.circleimageview.CircleImageView
import kotlinx.android.synthetic.main.single_recycle_item.view.*

class MyAdapter(private val playerList:List<Player>, private val listner:OnItmClick): RecyclerView.Adapter<MyAdapter.MyViewHolder>() {


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val view= LayoutInflater.from(parent.context).inflate(R.layout.single_recycle_item,parent,false)
        return MyViewHolder(view)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {

        holder.image.setImageResource(playerList[position].image)
        holder.name.text= playerList[position].Name
        holder.role.text= playerList[position].Roll
    }

    override fun getItemCount(): Int {
        return playerList.size
    }

    // override fun getItemCount()= playerList.size

    inner  class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView),View.OnClickListener,View.OnLongClickListener {

        val image: CircleImageView = itemView.singleRecycleImage
        val name: TextView = itemView.singleRecycleName
        val role: TextView = itemView.singleRecycleRole

        init {
            itemView.setOnClickListener(this)
            itemView.setOnLongClickListener(this)
        }
        override fun onClick(v: View?) {
            val position:Int = adapterPosition

            if (position != RecyclerView.NO_POSITION){
                listner.onItemClick(position)

            }
        }

        override fun onLongClick(v: View?): Boolean {
            val position:Int = adapterPosition

            if (position != RecyclerView.NO_POSITION){
                listner.onLongItemClick(position)

            }
          return false
        }

    }

    interface  OnItmClick{

        fun onItemClick(position:Int)
        fun onLongItemClick(position:Int)
    }
}